File: server/internal/models/chat_model.go
package models

import "time"

type SendMessageRequest struct {
	JobVacancyID int    `json:"job_vacancy_id" validate:"required"`
	Message      string `json:"message" validate:"required,min=1,max=5000"`
}

type ChatResponse struct {
	ID           int       `json:"id"`
	JobVacancyID int       `json:"job_vacancy_id"`
	SenderID     int       `json:"sender_id"`
	SenderType   string    `json:"sender_type"`
	SenderName   string    `json:"sender_name"`
	Message      string    `json:"message"` 
	IsRead       bool      `json:"is_read"`
	CreatedAt    time.Time `json:"created_at"`
}


type ChatThreadResponse struct {
	JobVacancyID  int       `json:"job_vacancy_id"`
	JobPosition   string    `json:"job_position"`
	CompanyName   string    `json:"company_name"`
	CompanyID     int       `json:"company_id"`
	UserName      string    `json:"user_name"`
	UserID        int       `json:"user_id"`
	LastMessage   string    `json:"last_message"`
	LastMessageAt time.Time `json:"last_message_at"`
	UnreadCount   int       `json:"unread_count"`
}

File: lib/data/model/chat_model.dart
class ChatMessage {
  final int id;
  final int jobVacancyId;
  final int senderId;
  final String senderType;
  final String senderName;
  final String message;
  final bool isRead;
  final DateTime createdAt;

  ChatMessage({
    required this.id,
    required this.jobVacancyId,
    required this.senderId,
    required this.senderType,
    required this.senderName,
    required this.message,
    required this.isRead,
    required this.createdAt,
  });

  factory ChatMessage.fromJson(Map<String, dynamic> json) {
    return ChatMessage(
      id: json['id'],
      jobVacancyId: json['job_vacancy_id'],
      senderId: json['sender_id'],
      senderType: json['sender_type'],
      senderName: json['sender_name'],
      message: json['message'],
      isRead: json['is_read'],
      createdAt: DateTime.parse(json['created_at']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'job_vacancy_id': jobVacancyId,
      'sender_id': senderId,
      'sender_type': senderType,
      'sender_name': senderName,
      'message': message,
      'is_read': isRead,
      'created_at': createdAt.toIso8601String(),
    };
  }
}

class SendMessageRequest {
  final String message;

  SendMessageRequest({required this.message});

  Map<String, dynamic> toJson() {
    return {
      'message': message,
    };
  }
}